include/zephyr/device.h

```diff
@@ -1058,8 +1058,47 @@ __syscall int device_init(const struct device *dev);
 
 DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)
 
+/** @brief Expands to the full type. */
+#define Z_DEVICE_API_TYPE(_class) _CONCAT(_class, _driver_api)
+
 /** @endcond */
 
+/**
+ * @brief Wrapper macro for declaring device API structs inside iterable sections.
+ *
+ * @param _class The device API class.
```

Comment:
```suggestion
 * @param _driver_api_name The name of the driver subsystem API. Usually this is the device class (adc, uart, i2c, ...) of the devices being exposed by this driver subsystem. If you define complex drivers that may wrap devices of different classes (rtc, stepper, ieee802154_radio), during migration (usb vs. usb_next) or if you implement a custom API for the same device class (adc_<vendor/project>), then we recommend to suffix the device class to designate the API variant uniquely.
```

@gmarull It turns out that this is the one thing where we have differing opinions. Can you see my point why we cannot always equate device classes and APIs precisely although they are of course very closely related?